pages/feed/index.tsx

```diff
@@ -101,10 +101,15 @@ const Feeds: WithPageLayout<HighlightSSRProps> = (props: HighlightSSRProps) => {
   useEffect(() => {
     const queryParams = new URLSearchParams(window.location.search);
     const newHighlight = queryParams.get("new");
+    const prURL = queryParams.get("prurl");
     const signInRequired = queryParams.get("signIn");
 
     if (newHighlight && signInRequired) {
-      signIn({ provider: "github", options: { redirectTo: `${window.location.origin}/feed?new=${newHighlight}` } });
+      let query = "";
+      for (const [key, value] of queryParams.entries()) {
+        query += `${key}=${value}&`;
+      }
+      signIn({ provider: "github", options: { redirectTo: `${window.location.origin}/feed?${query}` } });
```

Comment:
Refactor (blocking): Use `queryParams` built in `.toString()` which gets used implicitly when used in a template string. Another benefit to using it is it encodes the querystring values.

```suggestion
      signIn({ provider: "github", options: { redirectTo: `${window.location.origin}/feed?${queryParams}` } });
```